ports/rp2/machine_rtc.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * astrorafael port of RP2 Real Time Clock module, based on previous
+ * work by the MicroPython project
+ *
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020-2021 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// --------------------
+// MicroPython includes
+// --------------------
+
+#include "py/runtime.h"
+#include "py/mperrno.h"
+#include "py/mphal.h"
+#include "lib/utils/mpirq.h"
+
+// ----------------------
+// Pico specific includes
+// ----------------------
+
+#include "hardware/rtc.h"
+#include "pico/util/datetime.h"
+
+// ------------------
+// Exported constants
```

Comment:
There's no such thing as 'exporting' a PP define

Reply:
ok

Reply:
Regarding the number of commits, this means that I'll have to delete my fork and start from scratch, right?
Also I have to fix the commit message style.

Reply:
> this means that I'll have to delete my fork and start from scratch

No. There are a bunch of ways to do this. Given the number of commits: something like `git reset --mixed` will keep all your changes but erase all commits, so you can create a single new one.

Reply:
Finally I started from scratch and issued PR  #6929 . I tried git reset --mixed but wasn't sure about the results and didn't want to mess it up with the commits.
In the mean time, it seems another developer wants to PR (#6928) another implementation of the same port and functionality. 
Funny ...

Reply:
t seems that, somehow, updating my local & remote repos with upstream and doing git reset --mixed, made PR #6929 to close. I did not want to close it on purpose. Now there is a #6932, with all green lights. Sorry for all this mess and thank you for your support. It is also ok with me if the project chooses #6928.